docs/core-concepts/networking.md

```diff
@@ -137,11 +138,25 @@ The overlay network is established using the network addresses of the peer Nodes
 The overlay network mesh uses the Private network address of a peer node within the same Region; otherwise, the Public network address is used.
 Once the overlay network is started, the host Node's overlay network address is configured using [`weave expose`](https://www.weave.works/docs/net/latest/using-weave/host-network-integration/).
 
-The overlay network is powered by Weave Net, using [Weave's encrypted `sleeve` tunnels](https://www.weave.works/docs/net/latest/using-weave/security-untrusted-networks/) to form a flat Layer 2 network spanning all Grid Nodes and connected Containers.
+The overlay network, powered by Weave Net, is created to form a flat Layer 2 network spanning all Grid Nodes and connected Containers. Overlay network operates at three different modes depending on the environment and grid configuration, see below for their descriptions. **Selecting the fastest forwarding approach is automatic, and is determined on a connection-by-connection basis.**
 
-Alternatively, [Weave's Fast Datapath](https://www.weave.works/docs/net/latest/using-weave/fastdp/) can be used for traffic between Nodes within the Kontena Grid's [Trusted Subnets](https://kontena.io/docs/using-kontena/grids#grid-trusted-subnets).
+### Fast Datapath
+
+[Weave's Fast Datapath](https://www.weave.works/docs/net/latest/using-weave/fastdp/) can be used for traffic between Nodes within the Kontena Grid's [Trusted Subnets](https://kontena.io/docs/using-kontena/grids#grid-trusted-subnets).
 Using Trusted Subnets and Weave's Fast Datapath provides [improved performance](https://www.weave.works/weave-docker-networking-performance-fast-data-path/). The cost of this method is a lack of data plane encryption between Nodes.
 
+### Encrypted Datapath
+
+If no `trusted subnets` have been configured in a grid, the overlay network creates IPSec encrypted Fast Datapath connections between the peers.
+
+**NOTE:**
+Each encrypted dataplane packet is encapsulated into ESP, thus in some networks a firewall rule for allowing ESP traffic needs to be installed. E.g. Google Cloud Platform denies ESP packets by default. When upgrading from a version earlier than 1.2.0, if your host's network interface has a limit on packet size (the "MTU") smaller than 1496 bytes, you should reboot after upgrading to ensure encrypted fast datapath can work. For instance this applies to Google Cloud Platform, but is not necessary on AWS.
```

Comment:
Hmm... do we have something to link to for the GCP stuff? Maybe:

> The Google Cloud Platform [default firewall rules](https://cloud.google.com/compute/docs/networking#firewall_rules) do not allow IPSec ESP connections, and you must configure the firewall rules in order to use encrypted Weave `fastdp`.

Bonus points for actually testing Kontena 1.2.0 on GCP and including specific instructions here... separate PR :D